ContractsApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs

```diff
@@ -252,6 +252,30 @@ public async Task PatchContractThrowsConflictVersionError()
                 Times.Never());
         }
 
+        [Fact]
+        public async Task PatchContractThrowsErrorIfTryingToSuspendBlock()
+        {
+            var today = DateTime.Today;
+            var tomorrow = today.AddDays(1);
+            var currentContract = _fixture.Build<ContractDb>().With(x => x.VersionNumber, (int?) null).With(x => x.StartDate, (DateTime?) today).Create();
+
+            await InsertDataIntoDynamoDB(currentContract).ConfigureAwait(false);
+
+            var contractId = currentContract.Id;
+            var request = _fixture.Create<EditContractRequest>();
+            var suppliedVersion = 0;
+            request.HandbackDate = tomorrow;
+            request.SuspensionDate = tomorrow;
+            request.ContractManagement.AssetHierarchy = AssetHierarchy.Block;
+
+            Func<Task<UpdateEntityResult<ContractDb>>> func = async () =>
+                await _classUnderTest.PatchContract(contractId, request, It.IsAny<string>(), suppliedVersion).ConfigureAwait(false);
+
+            await func.Should().ThrowAsync<SuspendingBlockException>().WithMessage($"It is not possible to add a suspension to blocks");
+            _logger.VerifyExact(LogLevel.Debug, $"Calling IDynamoDBContext.SaveAsync to update id {contractId}",
```

Comment:
The way the logic flows in the gateway I think it's sufficient to test that the `SuspendingBlockException` expection is thrown.  Testing what the logger does here is not really reliable since that implementation might change at some point. For example the logging could be removed altogether from the gateway method and this verification would still be true.

Reply:
True, deleted log check.